components/resource_control/src/resource_limiter.rs

```diff
@@ -29,6 +29,12 @@ pub struct ResourceLimiter {
     limiters: [QuotaLimiter; ResourceType::COUNT],
```

Comment:
Does it have any benefit? I think using separated fields for cpu and io is more readable.

Reply:
With an array, we can index the limiter via `ResourceType as usize` without extra code, It's more convenient than separated field. Not sure if explicitly fields is better here.